demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,506 @@
+    TBlob<currentBlobType>& tblob = dynamic_cast<TBlob<currentBlobType>&>(inputBlob);
+    size_t batchSize =  tblob.getTensorDesc().getDims()[0];
+    numTop = static_cast<unsigned>(std::min<size_t>(size_t(numTop), tblob.size()));
+    
+    std::vector<std::vector<unsigned>> output(batchSize);
+    for (size_t i = 0; i < batchSize; i++) {
+        size_t offset = i * (tblob.size() / batchSize);
+        currentBlobType *batchData = tblob.data();
+        batchData += offset;
+        std::vector<unsigned> indices(tblob.size() / batchSize);
+        std::iota(std::begin(indices), std::end(indices), 0);
+        std::partial_sort(std::begin(indices), std::begin(indices) + numTop, std::end(indices),
+                          [&batchData](unsigned l, unsigned r) {
+                              return batchData[l] > batchData[r];
+                          });
+
+        indices.resize(numTop);
+        output[i] = std::move(indices);
+    }
+    
+    return output;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        // -----------------------------------------Read input images-----------------------------------------
+        std::vector<std::string> imageNames;
+        std::vector<cv::Mat> inputImages;
+        parseInputFilesArguments(imageNames);
+        size_t inputImagesCount = imageNames.size();
+        if (inputImagesCount == 0) throw std::runtime_error("No images provided");
+        std::sort(imageNames.begin(), imageNames.end());
+        for (size_t i = 0; i < inputImagesCount; i++) {
+            const std::string& name = imageNames[i];
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                inputImagesCount--;
+                i--;
+            } else {
+                inputImages.push_back(tmpImage);
+                imageNames[i] = name.substr(name.rfind('/') + 1);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Read image classes-----------------------------------------
+        std::vector<unsigned> classIndices;
+        if (!FLAGS_gt.empty()) {
+            std::map<std::string, std::string> classIndicesMap;
+            std::ifstream inputGtFile(FLAGS_gt);
+            if (!inputGtFile.is_open()) throw std::runtime_error("Can't open the ground truth file.");
+            while (true) {
+                std::string imageName;
+                std::string classIndex;
+                inputGtFile >> imageName >> classIndex;
+                if (inputGtFile.eof()) break;
+                classIndicesMap.insert({imageName.substr(imageName.rfind('/') + 1), classIndex});
```

Comment:
@IRDonch 

I discussed it with Wovchena. I think that as long as there are not much classification datasets with complex subfolder structure and parseInputFilesArguments() function doesn't check subfolders, we can omit all paths.

So, the GT file will contain only file names (without paths) and name uniqueness will be already guaranteed just because all files are in the same folder.

It will require updating GT file in CI and updating GT file format description in Readme.

Do you agree with this solution idea?

Reply:
> name uniqueness will be already guaranteed just because all files are in the same folder.

Not true, because the same file could be listed twice. So you should still make sure the names are unique.

> It will require updating GT file in CI

Question: where did that file come from to begin with?

Reply:
@IRDonch 

This file comes from ILSVRC2012_img_val dataset.
I use this dataset for testing locally and CI tests use the same dataset.

Reply:
If it comes from the dataset, then you shouldn't modify it. The users need to be able to run the tests, so the files should correspond to those found in the dataset. You'll have to instead ensure in the demo that path listed in the GT file refers to the current directory.